lib/flight_configuration/base_dsl.rb

```diff
@@ -231,9 +218,10 @@ def merge_sources(config)
         end
 
         # Apply the defaults
-        defaults.each do |key, value|
+        attributes.values.each do |attr|
+          key = attr[:name].to_s
           next if sources[key]
-          sources[key] = SourceStruct.new(key, nil, :default, value, config)
+          sources[key] = SourceStruct.new(key, nil, :default, attr[:default], config)
```

Comment:
Not a major problem, AFAICS this all works. However this line should probably be:

```
sources[key] = SourceStruct.new(key, nil, :default, nil, config)
```

The `value_before_type_cast` input is ignored for the `default` sources, so it doesn't make a difference either way. However current it implies the actual value could be a `proc`, as opposed to the result.